### PR TITLE
node/pkg/watcher: sui refactor plus tests

### PR DESCRIPTION
**Pull request by panoel**
This refactors runWithScissors() to call functions instead of running the function inline.
Also, adds a test file that can be run with **go run**
The diff in vscode is much nicer than the one here.

### CODE REVIEW

File: node/pkg/watchers/sui/watcher_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	fmt.Println("inspectBody had an error", err)
+	assert.Contains(tst, err.Error(), "Transaction hash is not 32 bytes")
+
+	hashBase := "This needs to be a 32 byte hash."
+	txd = base58.Encode([]byte(hashBase))
+	err = watcher.inspectBody(res)
+	require.NoError(tst, err)
+	require.Equal(tst, 1, len(msgChan))
+	obsv := <-msgChan
+	fmt.Println("Received msgChan msg...")
```
Comment (bruce-riley) [inline node/pkg/watchers/sui/watcher_test.go:108]: Don't really need all these print statements since assert will tell you the line number of an error.

**Review by bruce-riley — DISMISSED** [review]

File: node/pkg/watchers/sui/watcher_test.go (hunk 1 above)
Comment (panoel) [reply node/pkg/watchers/sui/watcher_test.go]: Fixed

File: node/pkg/watchers/sui/watcher.go [hunk 2]
```diff
@@ -222,12 +223,12 @@ func (e *Watcher) inspectBody(logger *zap.Logger, body SuiResult) error {
 
 	seq, err := strconv.ParseUint(*fields.Sequence, 10, 64)
 	if err != nil {
-		logger.Info("Sequence decode error", zap.String("Sequence", *fields.Sequence))
+		w.logger.Info("Sequence decode error", zap.String("Sequence", *fields.Sequence))
```
Comment (bruce-riley) [inline node/pkg/watchers/sui/watcher.go:226]: Should "decode errors" be info logs, or error logs?

**Review by bruce-riley — APPROVED** [review]

File: node/pkg/watchers/sui/watcher.go (hunk 2 above)
Comment (panoel) [reply node/pkg/watchers/sui/watcher.go]: Changed to Error

**Comment by panoel** [comment]
The SUI watcher ha been running without incident for quite some time.  This refactor is not necessary at this time.  It will be reevaluated should we need to address issues in the Sui watcher in the future.

**Pull request closed without merging**